src/main/java/com/metamx/common/StreamUtils.java

```diff
@@ -133,6 +139,7 @@ public static long copyWithTimeout(InputStream is, OutputStream os, long timeout
       os.write(buffer, 0, n);
       size += n;
     }
+    os.flush();
```

Comment:
I believe the flush should be left to the caller and not be part of this method. Whether the caller uses a buffered or unbuffered input stream should not be the concern of this method.